Source/Core/Gen8/Generators/WildGenerator8.cpp

```diff
@@ -18,26 +18,181 @@
  */
 
 #include "WildGenerator8.hpp"
+#include <Core/Enum/Game.hpp>
 #include <Core/Enum/Method.hpp>
+#include <Core/Parents/PersonalLoader.hpp>
 #include <Core/RNG/RNGList.hpp>
 #include <Core/RNG/Xorshift.hpp>
+#include <Core/Util/EncounterSlot.hpp>
 
-WildGenerator8::WildGenerator8(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, u8 genderRatio, const StateFilter &filter) :
-    WildGenerator(initialAdvances, maxAdvances, tid, sid, genderRatio, Method::Null, filter)
+WildGenerator8::WildGenerator8(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, const StateFilter &filter) :
+    WildGenerator(initialAdvances, maxAdvances, tid, sid, 0, Method::Null, filter)
 {
 }
 
+void WildGenerator8::setEncounterArea(const EncounterArea8 &encounterArea)
+{
+    this->encounterArea = encounterArea;
+}
+
 std::vector<WildState> WildGenerator8::generate(u64 seed0, u64 seed1) const
 {
     Xorshift rng(seed0, seed1);
     rng.advance(initialAdvances + offset);
 
-    RNGList<u32, Xorshift, 2, 0> rngList(rng);
-
     std::vector<WildState> states;
-    for (u32 cnt = 0; cnt <= maxAdvances; cnt++, rngList.advanceState())
+    for (u32 cnt = 0; cnt <= maxAdvances; cnt++, rng.next())
     {
-        
+        WildState state(initialAdvances + cnt);
+        Xorshift gen(rng);
+        u8 slotPercent = gen.next<0, 100>();
+        gen.advance(84);
+        switch (encounter)
+        {
+        case Encounter::Grass:
+            state.setEncounterSlot(EncounterSlot::bdspSlot(slotPercent, encounter));
+            if (!filter.compareEncounterSlot(state))
+            {
+                continue;
+            }
+
+            state.setLevel(encounterArea.calcLevel(state.getEncounterSlot()));
+            break;
+        case Encounter::Surfing:
+        case Encounter::OldRod:
+        case Encounter::GoodRod:
+        case Encounter::SuperRod:
+            state.setEncounterSlot(EncounterSlot::bdspSlot(slotPercent, encounter));
+            if (!filter.compareEncounterSlot(state))
+            {
+                continue;
+            }
+
+            state.setLevel(encounterArea.calcLevel(state.getEncounterSlot(), gen.next<0, 10000>()));
+            break;
+        default:
+            break;
+        }
+
+        PersonalInfo info = PersonalLoader::getPersonal(Game::BDSP, encounterArea.getPokemon()[state.getEncounterSlot()].getSpecie(), 0);
```

Comment:
Each Slot in EncounterArea8 has PersonalInfo attached. That should be used instead